tools/ci_build/build.py

```diff
@@ -2106,7 +2117,7 @@ def run_csharp_tests(source_dir, build_dir, use_cuda, use_openvino, use_tensorrt
     if use_cuda:
         macros += "USE_CUDA;"
     if enable_training_apis:
-        macros += "__TRAINING_ENABLED_NATIVE_BUILD__;"
+        macros += "__TRAINING_ENABLED_NATIVE_BUILD__;__ENABLE_TRAINING_APIS__"
```

Comment:
__TRAINING_ENABLED_NATIVE_BUILD__ -> This is used in test project as a signal to determine whether to run training tests. This is OFF by default.

__ENABLE_TRAINING_APIS__ -> This is used in the c# bindings to include training bindings. The training bindings are included by default, we keep this just so that we have an easy way to exclude them if necessary. The way these product and test projects are hooked, the test project is not picking up this macro, so I had to explicitly add it here. Ideally, we don't need to explicitly set this.